File: 3.ConvertDecimaltoBinary/CPP/src/DecimalToBinary.cpp
#include "../include/DecimalToBinary.hpp"



std::string DecimalToBinary::decimalToBinary(int decimal) {
    std::string binary;

    if (decimal == 0) {
        binary = "0";
    } else {
        while (decimal > 0) {
            int remainder = decimal % 2;
            binary = std::to_string(remainder) + binary;
            decimal /= 2;
        }
    }

    return binary;
}

File: 3.ConvertDecimaltoBinary/PYTHON/DecimalToBinary.py
# Python program to convert decimal to binary
# Visit https://venkys.io for more information

class DecimalToBinary:
    # Function to convert decimal to binary
    def decimal_to_binary(self, decimal):
        # Initialize an empty string to store the binary representation.
        binary = ""

        # Convert decimal to binary
        while decimal > 0:
            remainder = decimal % 2  # Calculate the remainder when decimal is divided by 2.
            binary = str(remainder) + binary  # Insert the remainder at the beginning of the binary string.
            decimal //= 2  # Update decimal by dividing it by 2.

        return binary  # Return the binary representation.

# Test the function
decimal = int(input("Enter a decimal number: "))
converter = DecimalToBinary()
binary = converter.decimal_to_binary(decimal)
print("Binary:", binary)
